### DISCUSSION TITLE
OpenAI API - context lenght limit problem!

### DISCUSSION THREAD

**Discussion by doomgrave** [oobabooga/text-generation-webui#4011] [Q&A]
Hi. im using Textgeneration + llama.cpp + OpenAI api + TheBloke/OpenBuddy-Llama2-13B-v11.1-GGUF that should have a Context lenght of 4096.

n_ctx it's already 4096, but whatever setting i use I get this error:
 $This model maximum context length is 2048 tokens. However, your messages resulted in over 2560 tokens and max_tokens is 2048.

I  must set some special parameter to use the full 4096 tokens?

**Comment by vampirismtrueblood** [comment]
Similar to this #3910

**Reply by jiuchen1986** [reply]
I'm hit by the same issue, and I don't think it's similar to #3910. This issue spit a Warning message rather than an exception explicitly from the OpenAI extension. And the workaround doesn't work in this case. My case is Textgeneration + autogptq + OpenAI api + TheBloke/Llama-2-13B-chat-GPTQ where the model is also claimed with 4096 context window. The fix for me is actually change the model config:

```
# config.json in model folder
{
  "architectures": [
    "LlamaForCausalLM"
  ],
  "bos_token_id": 1,
  "eos_token_id": 2,
  "hidden_act": "silu",
  "hidden_size": 5120,
  "initializer_range": 0.02,
  "intermediate_size": 13824,
  "max_position_embeddings": 2048, # changing this from 2048 -> 4096 resolves the warning
  "model_type": "llama",
  "num_attention_heads": 40,
  "num_hidden_layers": 40,
  "pad_token_id": 0,
  "rms_norm_eps": 1e-05,
  "tie_word_embeddings": false,
  "torch_dtype": "float16",
  "transformers_version": "4.30.2",
  "use_cache": true,
  "vocab_size": 32000,
  "quantization_config": {
    "bits": 4,
    "group_size": 32,
    "damp_percent": 0.01,
    "desc_act": true,
    "sym": true,
    "true_sequential": true,
    "model_name_or_path": null,
    "model_file_base_name": "model",
    "quant_method": "gptq"
  }
}
```
